packages/cicero-core/src/template.js

```diff
@@ -172,15 +178,60 @@ class Template {
         return hasher.digest('hex');
     }
 
+    /**
+     * signs a string made up of template hash and time stamp using private key derived
+     * from the keystore
+     * @param {String} [p12File] - encoded string of p12 keystore file
+     * @param {String} [passphrase] - passphrase for the keystore file
+     * @param {Number} [timeStamp] - timeStamp of the moment of signature is done
+     * @private
+     */
+    signTemplate(p12File, passphrase, timeStamp) {
```

Comment:
What happens if any of these values are null or undefined? Can you program defensively here, please?

Reply:
These parameters were checked before calling this method. Moreover this method is private and more has more of an auxiliary purpose so I had kept that method a bit loose. I'll add a check for these parameters in next commit.